apps/web/pages/signup.tsx

```diff
@@ -74,9 +75,10 @@ export default function Signup({ prepopulateFormValues, token, orgSlug, orgAutoA
     }
   };
 
-  const isOrgInviteByLink = orgSlug && !prepopulateFormValues?.username;
-
   const signUp: SubmitHandler<FormValues> = async (data) => {
+    if (data.username === undefined && isOrgInviteByLink && orgAutoAcceptEmail) {
+      data = { ...data, username: getOrgUsernameFromEmail(methods.getValues().email, orgAutoAcceptEmail) };
+    }
```

Comment:
This username that is inferred was not being added to the form data correctly before sending to the server